pkg/aggregator/context_resolver.go

```diff
@@ -16,14 +16,24 @@ import (
 
 // Context holds the elements that form a context, and can be serialized into a context key
 type Context struct {
-	Name string
-	Host string
-	tags *tags.Entry
+	Name          string
+	Host          string
+	compositeTags *metrics.CompositeTags
+	tags          *tags.Entry
+}
+
+func newContext(name string, host string, tags *tags.Entry) *Context {
+	return &Context{
+		Name:          name,
+		Host:          host,
+		tags:          tags,
+		compositeTags: metrics.CompositeTagsFromSlice(tags.Tags()),
```

Comment:
Why we do this when we create the context? I think we could do this on-demand in `Context#Tags`.

Reply:
I think `on-demand` makes sense if `Context#Tags` would be rarely called whereas I think it is called for every instance.

Reply:
Can't say I agree. Pre-creating it means we have to store 56 extra bytes per context (48 if you consider rounding up to a sizeclass). This is 70% more compared to current size (not the biggest part, but still can add up to tens of megabytes under high load).

Please also see my [other comment](https://github.com/DataDog/datadog-agent/pull/10460/files#r780215034) about embedding CompositeTags in the Serie. If we don't need to allocate, creating on-demand should as cheap as copying two slices.

Reply:
I understood `on-demand` as lazy initialization, but If I understand correctly, you suggest calling `NewCompositeTags` each time a `CompositeTags` is needed.
It is doable but it depends on https://github.com/DataDog/datadog-agent/pull/10460#discussion_r783042670.